src/github3/repos/repo.py

```diff
@@ -499,7 +539,82 @@ def contributors(self, anon=False, number=-1, etag=None):
+                         actions=None):
+        """Create a check run object on a commit
+
+        .. versionadded:: 1.3.0
+
+        :param str name:
+            (required), The name of the check
+        :param str head_sha:
+            (required), The SHA of the commit
+        :param str details_url:
+            (optional), The URL of the integrator's site that has the full
+            details of the check
+        :param str external_id:
+            (optional), A reference for the run on the integrator's system
+        :param str started_at:
+            (optional), ISO 8601 time format: YYYY-MM-DDTHH:MM:SSZ
+        :param str status:
+            (optional), ('queued', 'in_progress', 'completed')
+        :param str conclusion:
+            (optional), Required if you provide 'completed_at', or a
+            'status' of 'completed'. The final conclusion of the check.
+            ('success', 'failure', 'neutral', 'cancelled', 'timed_out',
+            'action_required')
+        :param str completed_at:
+            (optional), Required if you provide 'conclusion'. ISO 8601 time
+            format: YYYY-MM-DDTHH:MM:SSZ
+        :param dict output:
+            (optional), key-value pairs representing the output. Format:
+            {'title': 'string', 'summary', 'text, can be markdown', 'text':
+            'text, can be markdown', 'annotations': [{}], 'images': [{}]}
+        :param array actions:
+            (optiona), array of action objects. Object format is:
+            {'label': 'text', 'description', 'text', 'identifier', 'text'}
+        :returns:
+            the created check run
+        :rtype:
+            :class:`~github3.checks.CheckRun`
+        """
+        json = None
+        # TODO: Cleanse output dict, actions array
+        if name and head_sha:
+            data = {'name': name, 'details_url': details_url, 'external_id':
+                    external_id, 'started_at': started_at, 'status': status,
+                    'conclusion': conclusion, 'completed_at': completed_at,
+                    'output': output, 'actions': actions}
+            self._remove_none(data)
+            url = self._build_url('check-runs', base_url=self._api)
+            json = self._json(self._post(url, data=data), 201)
+        return self._instance_or_null(checks.CheckRun, json)
+
+    @decorators.requires_app_installation_auth
+    def create_check_suite(self, head_sha):
+        """Create a check suite object on a commit
+
+        .. versionadded:: 1.3.0
+
+        :param str head_sha:
+            The sha of the head commit.
+        :returns:
+            the created check suite
+        :rtype:
+            :class:`~github3.checks.CheckSuite`
+        """
+        json = None
+        if head_sha:
+            data = {'head_sha': head_sha}
+            self._remove_none(data)
+            url = self._build_url('check-suites', base_url=self._api)
+            json = self._json(self._post(url, data=data), 201)
```

Comment:
We need to add the preview `Accept` header here (`check.CheckSuite.CUSTOM_HEADERS`)